src/hexen/g_game.c

```diff
@@ -2093,6 +2095,12 @@ void G_DoPlayDemo(void)
         PlayerClass[i] = *demo_p++;
     }
 
+    if (playeringame[1] || M_CheckParm("-solo-net") > 0
```

Comment:
You can use M_ParmExists() for simplicity.

Reply:
Updated